packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1503,6 +1679,117 @@ export function attach(
     return changedKeys;
   }
 
+  function canObjectBeMemoized(prev: any, next: any): boolean {
+    if (prev == null || next == null) {
+      return false;
+    }
+
+    // We can't report anything meaningful for hooks changes.
+    if (isHooksChange(next)) {
+      return false;
+    }
+
+    const keys = new Set([...Object.keys(prev), ...Object.keys(next)]);
+    return Array.from(keys).every(key => isEqual(prev[key], next[key]));
+  }
+
+  function isFunctionOrComponent(value: any) {
+    return typeof value === 'function' || isValidElement(value);
+  }
+
+  /**
+   * Scan through a fiber's parent to find function & component props that
+   * weren't passed through props or created from a callback/memo.
+   */
+  function getNonMemoizedProps(
+    fiber: Fiber,
+    changedPropKeys: Array<string> | null,
+  ): Array<string> | null {
+    const debugOwner = fiber._debugOwner;
+    if (
+      changedPropKeys == null ||
+      debugOwner == null ||
+      // Ignore memo parents since they just pass the props through.
+      getTypeSymbol(debugOwner.type) === MEMO_SYMBOL_STRING
+    ) {
+      return null;
+    }
+
+    // Filter props that are passed through from the parent
+    const createdProps = changedPropKeys
+      // Ignore the children prop.
+      .filter(prop => prop !== 'children')
+      .filter(
+        prop => debugOwner.memoizedProps[prop] !== fiber.memoizedProps[prop],
+      )
+      // For now, only filter functions and elements.
+      .filter(prop => isFunctionOrComponent(fiber.memoizedProps[prop]));
+
+    if (createdProps.length === 0) return null;
+
+    const elementType = getElementTypeForFiber(debugOwner);
+
+    let memoValues = [];
+    if (elementType === FunctionComponent) {
+      memoValues = getMemoValues(debugOwner);
+    } else if (elementType === ClassComponent) {
+      memoValues = getStaticClassValues(debugOwner);
+    }
+
+    const memoValueSet = new Set([memoValues]);
+
+    // We can check our hook values to find functions not
+    // created from useCallback/useMemo or static properties.
+    return createdProps.filter(
+      prop => !memoValueSet.has(fiber.memoizedProps[prop]),
+    );
+  }
+
+  const REACT_LIFECYCLE_FUNCTIONS = new Set([
+    'constructor',
+    'shouldComponentUpdate',
+    'render',
+    'getSnapshotBeforeUpdate',
+    'componentDidMount',
+    'UNSAFE_componentWillUpdate',
+    'UNSAFE_componentWillReceiveProps',
+    'componentWillUnmount',
+    'componentDidCatch',
+  ]);
+
+  // Returns functions and components that exist on the class instance.
+  function getStaticClassValues(fiber: Fiber) {
+    const propertyNames = Object.getOwnPropertyNames(
+      fiber.stateNode.constructor.prototype,
+    );
+
+    const staticFunctions = propertyNames.filter(
+      propertyName => !REACT_LIFECYCLE_FUNCTIONS.has(propertyName),
+    );
+
+    return [
+      ...staticFunctions,
+      ...Object.values(fiber.stateNode).filter(isFunctionOrComponent),
```

Comment:
I don't understand why you're iterating over `fiber.stateNode` – to grab instance methods and props? Can you explain why this is relevant?

Instance methods don't cause a component to re-render, only props/state.